app/scripts/controllers/transactions/index.js

```diff
@@ -874,26 +871,13 @@ export default class TransactionController extends EventEmitter {
 
     // For 'rpc' we need to use the same basic configuration as mainnet, since
     // we only support EVM compatible chains, and then override the
-    // name, chainId and networkId properties. This is done using the
+    // name, and chainId properties. This is done using the
     // `forCustomChain` static method on the Common class.
     const chainId = parseInt(this._getCurrentChainId(), 16);
-    const networkStatus = this.getNetworkStatus();
-    const networkId = this.getNetworkId();
 
     return Common.custom({
       name,
       chainId,
-      // It is improbable for a transaction to be signed while the network
-      // is loading for two reasons.
-      // 1. Pending, unconfirmed transactions are wiped on network change
-      // 2. The UI is unusable (loading indicator) when network is loading.
-      // setting the networkId to 0 is for type safety and to explicity lead
-      // the transaction to failing if a user is able to get to this branch
-      // on a custom network that requires valid network id. I have not ran
-      // into this limitation on any network I have attempted, even when
-      // hardcoding networkId to 'loading'.
-      networkId:
-        networkStatus === NetworkStatus.Available ? parseInt(networkId, 10) : 0,
```

Comment:
Are we not losing functionality here if we're no longer concerned with the network status?

Should we use the `_getChainId` method instead as that already factors the `NetworkStatus`, converts to a decimal, and even returns zero as this removed logic did.

Reply:
afaik, networkId isn't used in any part of the unsignedTx generation.

I see that in the next version of @ethereumjs/common (which we currently are behind) the common.Custom class's networkId param is [explicitly checked](https://github.com/ethereumjs/ethereumjs-monorepo/blob/acd1bbbcb436f8e3f137e06b7f85271424e5cd09/packages/common/src/common.ts#L271C8-L271C8) even though I don't see it being used to [build the unsigned transaction](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/transactionFactory.ts#L29C17-L29C27), which is what I expect. I don't understand why networkId on the Custom class is being enforced now. I guess we can just pass any value for networkId here. Maybe I'm missing something? If network ID is truly required by @ethereumjs libs for some reason, this could also force us to keep it around.

Reply:
We should be okay here